util/src/file.rs

```diff
@@ -1,17 +1,18 @@
 use anyhow::ensure;
 use std::fs::File;
 use std::io::Read;
+use std::os::unix::fs::OpenOptionsExt;
 use std::result::Result;
 use std::{fs::OpenOptions, path::Path};
 
 /// Open a file writable
-pub fn fopen_w<P: AsRef<Path>>(path: P) -> std::io::Result<File> {
-    OpenOptions::new()
-        .read(false)
-        .write(true)
-        .create(true)
-        .truncate(true)
-        .open(path)
+pub fn fopen_w<P: AsRef<Path>>(path: P, secret: bool) -> std::io::Result<File> {
```

Comment:
Can we refactor to fopen_w_public and fopen_w_secret? Or use a `enum Visibility { Public, Secret }`?

Reply:
Reworked with an `enum`.